judge.pl

```diff
@@ -303,6 +308,27 @@ sub validate_test {
     $sp_report->ok;
 }
 
+sub check_input_hash {
+    my ($pid, $rank, $filename) = @_;
+
+    eval { require Authen::Passphrase::SaltedDigest; 1; } or return;
+
+    my $data = $fu->read_lines($filename);
+    my $input = join "", @$data;
```

Comment:
If this is intended for normalization, I disagree.
Some separator must be preserved, otherwise `1\n2` and `12` would give same digest. A single `LF` maybe?
Also, use single quoted unless interpolating.